protocol/images/server.go

```diff
@@ -0,0 +1,108 @@
+package images
+
+import (
+	"context"
+	"database/sql"
+	"net"
+	"net/http"
+
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/protocol/identity/identicon"
+)
+
+type messageHandler struct {
+	db     *sql.DB
+	logger *zap.Logger
+}
+
+type identiconHandler struct {
+	logger *zap.Logger
+}
+
+func (s *identiconHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	pks, ok := r.URL.Query()["publicKey"]
+	if !ok || len(pks) == 0 {
+		s.logger.Error("no publicKey")
+		return
+	}
+	pk := pks[0]
+	image, err := identicon.Generate(pk)
+	if err != nil {
+		s.logger.Error("could not generate identicon")
+	}
+
+	w.Header().Set("Content-Type", "image/png")
```

Comment:
Would it make sense to add headers like these so the browser keeps the identicons in cache?:
```go
w.Header().Set("Cache-Control", "max-age:290304000, public")
w.Header().Set("Expires", time.Now().AddDate(60, 0, 0).Format(http.TimeFormat))
```